dtoa.c

```diff
@@ -1533,12 +1533,18 @@ ThInfo {
 set_max_dtoa_threads(unsigned int n)
 {
 	size_t L;
+	ThInfo *newTI1;
 
 	if (n > maxthreads) {
 		L = n*sizeof(ThInfo);
 		if (TI1) {
-			TI1 = (ThInfo*)REALLOC(TI1, L);
-			memset(TI1 + maxthreads, 0, (n-maxthreads)*sizeof(ThInfo));
+			newTI1 = (ThInfo*)REALLOC(TI1, L);
```

Comment:
Would you please tell me the url of this dtoa.c?

Reply:
This is based on
https://portal.ampl.com/~dmg/netlib/fp/dtoa.c
https://portal.ampl.com/~dmg/netlib/fp/changes

For this particular change you highlighted, I wrote it myself (see commit [026466c](https://github.com/openresty/lua-cjson/pull/84/commits/026466cd0f51db95508b4cf353d3f9d936e529dd)), as it was an issue reported in CI by cppcheck.